lib/kubeclient_next/client/rest_client.rb

```diff
@@ -17,10 +17,23 @@ def initialize(config:, context:, path: "")
         hardcoded_auth
       end
 
+      def post(sub_path = "", data:)
+        byebug
+        connection.post(formatted_uri(host, path, sub_path), data, { "Content-Type" => "application/json" })
+      end
+
```

Comment:
These methods expect data to simply be a string. For now, we are hardcoding `application/json` type, but in the future we can simply have that as the default and allow overriding/addition of custom headers. For the higher level `Client` API, we proactively call `JSON::dump` on the data so that users only have to worry about hashes (and/or ideally anything that can call `to_h` for when we deal with custom objects)